File: apps/backend-go/internal/core/domain/banner/valueobject/segment_tiers.go
package valueobject

// PredefinedSegmentTiers contains all valid segment tier values
// This is config-based for flexibility - can be easily modified without database changes
var PredefinedSegmentTiers = []string{
	"silver",
	"gold",
	"diamond",
	"platinum",
}

// ValidateSegmentTiers checks if all provided tiers are in the predefined list
func ValidateSegmentTiers(tiers []string) bool {
	// Create a map for O(1) lookup
	tierMap := make(map[string]bool)
	for _, tier := range PredefinedSegmentTiers {
		tierMap[tier] = true
	}

	// Check if all provided tiers are valid
	for _, tier := range tiers {
		if !tierMap[tier] {
			return false
		}
	}

	return true
}

File: apps/web/src/config/segmentTiers.ts
export const SEGMENT_TIERS = ["silver", "gold", "diamond", "platinum"] as const;

export type SegmentTier = (typeof SEGMENT_TIERS)[number];

export const SEGMENT_TIER_LABELS: Record<SegmentTier, string> = {
  silver: "เงิน",
  gold: "ทอง",
  diamond: "เพชร",
  platinum: "แพลตินัม",
};
